_studio/shared/umc/codec/jpeg_common/include/bitstreamin.h

```diff
@@ -56,9 +57,9 @@ class CBitStreamInput
   JERRCODE ReadDword(int* dword);
 
 protected:
-  CBaseStreamInput* m_in;
+  std::shared_ptr<CBaseStreamInput> m_in;
```

Comment:
from context below I believe you can use unique_ptr, not shared

Reply:
@mgonchar , seems we can't use unique_ptr - object under pointer should be changed in the function body after sent pointer to m_decBase. Look at https://github.com/Intel-Media-SDK/MediaSDK/blob/3d35d1c0be7453cffa62612961233801c033e36b/_studio/shared/umc/codec/jpeg_dec/src/umc_mjpeg_mfx_decode_hw.cpp#L257

Reply:
@akokoshn, didn't get why exactly we can't use unique_ptr in that code?

Reply:
@mgonchar , how we can call in->Seek(), after send in to SetSource() (move ownership), if in - unique_ptr?

Reply:
@akokoshn I see... ok, in general you are right, we can't do so. But this code breaks responsibility principle anyway - since we transferred ownership of CBaseStreamInput (even if it is a raw pointer when we setting source we mean that since that moment m_decBase is responsible for all operations with this input stream (btw thats why unique_ptr fits this model so well)), but after than we manipulate this pointer from outside for some reason. I see that there is a method CBitStreamInput::Seek, which supposed to be used for file pointer seeking, but for some reason it wasn't used. CBitStreamInput::Seek call is not equivalent to existing code, so it looks strange why it wasn't used. Could you check if we really need to operate with CBaseStreamInput objeect directly or we should call CBitStreamInput::Seek instead?

Reply:
@mgonchar , we haven't interface for call CBitStreamInput::Seek through CJPEGDecoderBase. and CBitStreamInput::Seek is not same as CMemBuffInput::Seek()